File: sdk/src/main/java/com/meniga/sdk/webservices/requests/GetChallenges.java
package com.meniga.sdk.webservices.requests;

import java.util.HashMap;
import java.util.Map;

/**
 * Copyright 2017 Meniga Iceland Inc.
 */
public class GetChallenges extends QueryRequestObject {

	public boolean includeExpired;
	public boolean excludeSuggested;
	public boolean excludeAccepted;

	@Override
	public Map<String, String> toQueryMap() {
		Map<String, String> map = new HashMap<>();
		map.put("includeExpired", Boolean.toString(includeExpired));
		map.put("excludeSuggested", Boolean.toString(excludeSuggested));
		map.put("excludeAccepted", Boolean.toString(excludeAccepted));
		return map;
	}

	@Override
	public long getValueHash() {
		int result = (includeExpired ? 1 : 0);
		result = 31 * result + (excludeSuggested ? 1 : 0);
		result = 31 * result + (excludeAccepted ? 1 : 0);
		return result;
	}
}

File: sdk/src/main/java/com/meniga/sdk/webservices/challenge/GetChallenges.kt
/*
 * Copyright 2017-2018 Meniga Iceland Inc.
 */
package com.meniga.sdk.webservices.challenge

import com.meniga.sdk.webservices.requests.QueryRequestObject

data class GetChallenges(
        @JvmField var includeExpired: Boolean = false,
        @JvmField var excludeSuggested: Boolean = false,
        @JvmField var excludeAccepted: Boolean = false,
        @JvmField var includeDisabled: Boolean = false
) : QueryRequestObject() {

    override fun toQueryMap(): Map<String, String> =
            mutableMapOf<String, String>().also { queryMap ->
                queryMap["includeExpired"] = includeExpired.toString()
                queryMap["excludeSuggested"] = excludeSuggested.toString()
                queryMap["excludeAccepted"] = excludeAccepted.toString()
                queryMap["includeDisabled"] = includeDisabled.toString()
            }

    override fun getValueHash(): Long = hashCode().toLong()
}
